jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -144,8 +146,19 @@ def vue_unit_conversion(self, *args, **kwargs):
 
         # Uncertainty converted to new flux units
         if self.spectrum.uncertainty is not None:
-            temp_uncertainty = self.spectrum.uncertainty.quantity.to(u.Unit(set_flux_unit.unit),
+            unit_exp = unit_exponents.get(self.spectrum.uncertainty.__class__, 1)
+
+            try:
+                temp_uncertainty = self.spectrum.uncertainty.quantity.to(u.Unit(set_flux_unit.unit)**unit_exp,
```

Comment:
instead of the try/except, might it work to just directly do what's happening in the ``except`` clause?  Then in ``except`` instead you just drop the uncertainties and issue a warning/snackbar message about dropping uncertainties due to incompatible types or something